sourced/ml/core/utils/bblfsh.py

```diff
@@ -1,19 +1,66 @@
 from distutils.version import StrictVersion
+from typing import List, Sequence
+
+from bblfsh.compat import CompatBblfshClient as BblfshClient
+from bblfsh.compat import CompatNodeIterator, Node
 
-from bblfsh.client import BblfshClient
 
 BBLFSH_VERSION_LOW = "2.2"
-BBLFSH_VERSION_HIGH = "3.0"
+BBLFSH_VERSION_HIGH = "4.0"
 
 
 def check_version(host: str = "0.0.0.0", port: str = "9432") -> bool:
     """
     Check if the bblfsh server version matches module requirements.
 
-    :param host: bblfsh server host
-    :param port: bblfsh server port
-    :return: True if bblfsh version specified matches requirements
+    :param host: bblfsh server host.
+    :param port: bblfsh server port.
+    :return: True if bblfsh version specified matches requirements.
     """
     # get version and remove leading 'v'
     version = StrictVersion(BblfshClient("%s:%s" % (host, port)).version().version.lstrip("v"))
     return StrictVersion(BBLFSH_VERSION_LOW) <= version < StrictVersion(BBLFSH_VERSION_HIGH)
+
+
+def iterate_children(nodes: Sequence[Node]) -> CompatNodeIterator:
```

Comment:
Let's have our `Node` class extension, with blackjack and whores.

The best would be to add these helper functions directly to the Node class in the Babelfish client. Can you please PR there.

Reply:
On halt till I'm sure I am replicating the old API correctly.

Reply:
@zurk All the breaking changes are handled using the functions in this file. This way you can quickly see in pycharm all the places where the new bblfsh version breaks.